source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -384,27 +383,42 @@ void CacheFilter::processSuccessfulValidation(Http::ResponseHeaderMap& response_
     return Http::HeaderMap::Iterate::Continue;
   });
 
-  if (should_update_cached_entry) {
-    // TODO(yosrym93): else the cached entry should be deleted.
-    // Update metadata associated with the cached response. Right now this is only response_time;
-    const ResponseMetadata metadata = {time_source_.systemTime()};
-    cache_.updateHeaders(*lookup_, response_headers, metadata);
-  }
+  // Update metadata associated with the cached response. Right now this is only response_time;
+  const ResponseMetadata metadata = {time_source_.systemTime()};
+  cache_.updateHeaders(*lookup_, response_headers, metadata);
 
   // A cache entry was successfully validated -> encode cached body and trailers.
   encodeCachedResponse();
 }
 
 // TODO(yosrym93): Write a test that exercises this when SimpleHttpCache implements updateHeaders
-bool CacheFilter::shouldUpdateCachedEntry(const Http::ResponseHeaderMap& response_headers) const {
-  ASSERT(isResponseNotModified(response_headers),
-         "shouldUpdateCachedEntry must only be called with 304 responses");
-  ASSERT(lookup_result_, "shouldUpdateCachedEntry precondition unsatisfied: lookup_result_ "
+bool CacheFilter::isSuccessfulValidation(const Http::ResponseHeaderMap& response_headers) const {
+  ASSERT(lookup_result_, "isSuccessfulValidation precondition unsatisfied: lookup_result_ "
                          "does not point to a cache lookup result");
   ASSERT(filter_state_ == FilterState::ValidatingCachedResponse,
-         "shouldUpdateCachedEntry precondition unsatisfied: the "
+         "isSuccessfulValidation precondition unsatisfied: the "
          "CacheFilter is not validating a cache lookup result");
+  // Should only update for 304 responses
```

Comment:
Maybe this could say: "If the response has not changed then the cache entry is valid"?

Reply:
Changed to "If the response has changed then the cache entry is not valid", because `isResponseNotModified` only checks for the 304 status code. It's insufficient to determine the validity.